### DISCUSSION TITLE
concurrency and CPU allocation

### DISCUSSION THREAD

**Discussion by aon108** [hibiken/asynq#349] [Q&A]
Could you provide any insights to the correct concurrency setting in a fixed resource environment like kubernetes?
For example:
Say I have a worker that is allocated 0.5CPU and performs some I/O based tasks like taking the task payload and writing to a DB, Im assuming that setting to concurrency of10 for example should bottleneck anything.

On the the other hand some other worker instances perform heavy CPU based tasks like data analysis etc. I think I should be setting the concurrency to 1, as i think the go routines will try steal cpu from each other.

Any advice would be much appreciated. Thank you

**Comment by hibiken** [comment]
I think it's difficult to give precise numbers since there're too many variables involved (e.g. OS, network bandwidth, etc).
But I guess in general, if a worker is processing mostly IO bound tasks, you'd get better performance if you have high concurrency since most of the time your workers are waiting on IO. And if a worker is processing mostly CPU bound tasks, you may need to adjust the concurrency to match the number of cores you have. But every program is different, so I would benchmark with different settings and find the numbers that work to get the most out of your systems :)
